### PR TITLE
Fix hash serialized calculation 

### PR DESCRIPTION
**Pull request by Naviabheeman**
Fix issue #289. 

This bug is fixed by using a parenthesis around _'second.fCoinBase'_ expression like in _TxInUndoSerializer_. It does not matter whether we use `* 2` or `<< 1 `. 

- Bitcoin moved away from VARINT and used `static_cast<uint32_t>`. I think we can keep VARINT as it gives the most compact encoding.
- Like Bitcoin `hash_serialized_2` is changed to `hash_serialized_3` to differentiate from the old hash output.

 For a coin with height **437**, the stream contents after pushing each expressions wasps follows:
 
| expression | output |
|--|--|
| VARINT(outputs.begin()->second.nHeight * 2 + outputs.begin()->second.fCoinBase ? 1u : 0u); |01 |
| VARINT(outputs.begin()->second.nHeight * 2 + (outputs.begin()->second.fCoinBase ? 1u : 0u)); | 856b|
| VARINT((outputs.begin()->second.nHeight << 1) + (outputs.begin()->second.fCoinBase ? 1u : 0u)); | 856b |
| VARINT((outputs.begin()->second.nHeight << 1)  + outputs.begin()->second.fCoinBase ? 1u : 0u); | 01 |
| uint32_t(outputs.begin()->second.nHeight * 2 + outputs.begin()->second.fCoinBase ? 1u : 0u); | 31 |
| uint32_t(outputs.begin()->second.nHeight * 2 + (outputs.begin()->second.fCoinBase ? 1u : 0u)); | 383735 |

### CODE REVIEW

File: test/functional/feature_dbcrash.py [hunk 1]
```diff
@@ -61,7 +61,7 @@ def set_test_params(self):
         self.node2_args = ["-dbcrashratio=24", "-dbcache=16"] + self.base_args
 
         # Node3 is a normal node with default args, except will mine full blocks
-        self.node3_args = ["-blockmaxweight=4000000"]
+        self.node3_args = ["-blockmaxsize=1000000"]
```
Comment (azuchi) [inline test/functional/feature_dbcrash.py:64]: Why this change needs?

File: test/functional/feature_fee_estimation.py [hunk 2]
```diff
@@ -134,8 +134,8 @@ def setup_network(self):
         which we will use to generate our transactions.
         """
         self.add_nodes(3, extra_args=[["-maxorphantx=1000", "-whitelist=127.0.0.1"],
-                                      ["-blockmaxweight=68000", "-maxorphantx=1000"],
-                                      ["-blockmaxweight=32000", "-maxorphantx=1000"]])
+                                      ["-blockmaxsize=17000", "-maxorphantx=1000"],
+                                      ["-blockmaxsize=8000", "-maxorphantx=1000"]])
         # Use node0 to mine blocks for input splitting
```
Comment (azuchi) [inline test/functional/feature_fee_estimation.py:139]: Why this change needs?

File: test/functional/feature_pruning.py [hunk 3]
```diff
@@ -127,7 +127,7 @@ def reorg_test(self):
         # Reboot node 1 to clear its mempool (hopefully make the invalidate faster)
         # Lower the block max size so we don't keep mining all our big mempool transactions (from disconnected blocks)
         self.stop_node(1)
-        self.start_node(1, extra_args=["-maxreceivebuffer=20000","-blockmaxweight=20000", "-checkblocks=5"])
+        self.start_node(1, extra_args=["-maxreceivebuffer=20000","-blockmaxsize=5000", "-checkblocks=5"])
```
Comment (azuchi) [inline test/functional/feature_pruning.py:130]: Why this change needs?

File: test/functional/feature_dbcrash.py [hunk 4]
```diff
@@ -218,7 +218,7 @@ def run_test(self):
 
         # Start by creating a lot of utxos on node3
         initial_height = self.nodes[3].getblockcount()
-        utxo_list = create_confirmed_utxos(self.nodes[3].getnetworkinfo()['relayfee'], self.nodes[3], 5000, self.signblockprivkey)
+        utxo_list = create_confirmed_utxos(self.nodes[3].getnetworkinfo()['relayfee'], self.nodes[3], 5000, self.signblockprivkey_wif)
```
Comment (azuchi) [inline test/functional/feature_dbcrash.py:221]: Why this change needs?

File: test/functional/feature_pruning.py (hunk 3 above)
Comment (Naviabheeman) [reply test/functional/feature_pruning.py:130]: We removed ```blockmaxweight``` argument from tapyrus. But these tests hadn't been updated. They are some long running scripts listed as ```EXTENDED_SCRIPTS```. These are not run by default by the CI. It has to be enabled in ```configure``` So I did not notice them until now. 
I wanted this change to be a separate PR after checking the running time. But had added them to PR unintentionally.

File: test/functional/feature_dbcrash.py (hunk 4 above)
Comment (Naviabheeman) [reply test/functional/feature_dbcrash.py:221]: This change should have been done earlier when all 'generate' rpcs were changed.  
 They are some long running scripts listed as EXTENDED_SCRIPTS. These are not run by default by the CI. It has to be enabled in configure So I did not notice them until now.
I wanted this change to be a separate PR after checking the running time. But I had to add them to this PR as this test needs it to execute correctly.

File: test/functional/feature_pruning.py (hunk 3 above)
Comment (azuchi) [reply test/functional/feature_pruning.py:130]: Since multiple changes should not be included in one PR, please delete it from this PR and create new PR.

File: test/functional/feature_dbcrash.py (hunk 4 above)
Comment (azuchi) [reply test/functional/feature_dbcrash.py:221]: Create another PR as above.

Comment (Naviabheeman) [reply test/functional/feature_dbcrash.py:221]: This PR is now clean i.e. only hash serialized bug fix and tis tests are present. Test case 'feature_dbcrash.py' is not executed in the CI now.
I am in the process of fixing the extended functional tests. This will be in a separate PR. After that one is merged all python tests will run in the CI every time.

**Review by azuchi — APPROVED** [review]
